lib/services/google_analytics.rb

```diff
@@ -0,0 +1,63 @@
+require 'google/apis/analyticsreporting_v4'
+require 'googleauth'
+
+module Services
+  class GoogleAnalytics
+    include Google::Apis::AnalyticsreportingV4
+    include Google::Auth
+
+    def client(scope: "https://www.googleapis.com/auth/analytics.readonly")
```

Comment:
What about extracting a class for the client? The service knows how to create the client, and it might not be necessary:

```ruby
def client
  @client ||= GoogleAnalytics::Client.new
end
```